### PR TITLE
feat(Tile): New component for Tile

### PR DESCRIPTION
**Pull request by ayeshakmaz**
Summary:

New Tile component for detailed items with actions
Storybook link to QA: https://5dcc95c567dd4d002006b897-sdoxpxuxjg.chromatic.com/?path=/story/tile--interactive-content-is-a-button

Screen Shot 2020-05-27 at 7 26 58 PM

JIRA:

[UXENG-78](https://optimizely.atlassian.net/browse/UXENG-78)

### CODE REVIEW

File: src/components/ReferenceCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import Button from '../Button';
+import OverlayWrapper from '../OverlayWrapper';
+import Popover from '../Popover';
+import Icon from 'react-oui-icons';
+
+import { greyDark } from '../../tokens/forimport/index.es';
+
+const renderCardActions = (items: (string | React.ReactElement)[]) => {
+  let actions = Array<React.ReactElement>(3);
+  items.forEach((item) => {
```
[the remark this answers is not in the document]
Comment (ayeshakmaz) [reply src/components/ReferenceCard/index.tsx:13]: Oh great point, I like the idea of just splitting status out in a separate prop, and will add the warning, thank you!

**Comment by ayeshakmaz** [comment]
Thank you for reviewing @davemn! Quite a few updates and renaming work have happened since, would you mind taking another look?

File: src/components/ButtonIcon/index.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ import classNames from 'classnames';
 import { FILL_COLOR_MAP } from '../../utils/accessibility';
 
 const ButtonIcon = ({
+  buttonRef,
```
[the remark this answers is not in the document]
Comment (kwalker3690) [reply src/components/ButtonIcon/index.js:8]: Although I do have to say I agree with the sentiment of this q on SO https://stackoverflow.com/questions/58578570/value-of-using-react-forwardref-vs-custom-ref-prop, which is that I don't really see why using ref forwarding is any more advantageous than a named prop, seems more like just the preferred approach acc to react standards? So I'll leave it up to you if you want to rewrite or not

File: src/components/Tile/Tile.story.js [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+              name="Delta"
+              description="ID:2468"
+              onTileClick={ action('onTileClick') }
+              isSelected={ boolean('isSelected', false) }
+              isDraggable={ boolean('isDraggable', false) }
+              hasWarning={ boolean('hasWarning', false) }
+              warningTitle={ text('warningText', 'This is a warning') }
+              warningContent={ <p>Warning details</p> }
+              status="Running"
+              dropdownItems={ [
```
Comment (kwalker3690) [inline src/components/Tile/Tile.story.js:157]: nit, should this prop be called 'actionItems'?

File: src/components/ButtonIcon/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ayeshakmaz) [reply src/components/ButtonIcon/index.js:8]: I think given that we use this same pattern for Button, I'll keep buttonRef prop to help with consistency. This way all our Buttons take a buttonRef prop, which should help build familiarity for users

File: src/components/Tile/Tile.story.js (hunk 3 above)
Comment (ayeshakmaz) [reply src/components/Tile/Tile.story.js:157]: I think dropdownItems helps make it extra clear that these items can only be surfaced through the ellipses dropdown, whereas we've been referring to all the different button icons as the Action items (edit, delete, copy, etc.)

**Pull request merged**